internal/result/results_test.go

```diff
@@ -260,3 +253,61 @@ func TestIter(t *testing.T) {
 		})
 	}
 }
+
+const totalResults = 10e6
```

Comment:
nit: To test this between 1 million and 10 million, you modified this constant, right? Consider a table-driven benchmark (https://go.dev/blog/subtests).

Running this at 10m makes it really clear why one is fast and one is slow.

Reply:
https://github.com/GoogleCloudPlatform/testgrid/pull/973